actionpack/test/dispatch/request/json_params_parsing_test.rb

```diff
@@ -38,10 +38,13 @@ def teardown
   end
 
   test "does not parse unregistered media types such as application/vnd.api+json" do
-    assert_parses(
-      {},
-      "{\"person\": {\"name\": \"David\"}}", "CONTENT_TYPE" => "application/vnd.api+json"
-    )
+    actual = "{\"person\": {\"name\": \"David\"}}"
+    headers = { "CONTENT_TYPE" => "application/vnd.api+json" }
+    assert_raise(ActionController::UnsupportedMediaType) {
+      with_test_routing do
+        post "/parse", params: actual, headers: headers
+      end
```

Comment:
Right now this, besides not raising an exception, returns an empty hash `{}`, and I haven't yet found where that's coming from.

This test may actually be more appropriate in a mime-responds / mime negotiation test file since it is not really about params parsing, at least in the form in this PR.